pubtools/pulplib/_impl/client/client.py

```diff
@@ -179,6 +182,42 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def upload_one_file(
+        self,
+        file_name,
+        repo_id,
+        type_id,
+        unit_key,
+        unit_metadata=None,
+        override_config=None,
+    ):
+        """Including four steps uploading a file:
+            1. request upload id from pulp
+            2. upload contents to pulp
+            3. import the uploaded content to wanted repo
+            4. delete the reuqest id
+        """
+        upload_id = self._request_upload().result()["upload_id"]
```

Comment:
You've written this all in blocking style, which means the current thread can't do anything else while waiting for the upload to be created, the import to finish, etc.  That means for example if someone needs to upload 10,000 files with this library, it'll be unnecessarily slow as it uploads and imports one file completely, then the next, then the next and so on, unable to handle any files concurrently.

Can you please ensure it's all in non-blocking style like the other methods?  It should be written in such a way that the call immediately returns a Future before any requests to Pulp have happened.

You probably want to use `f_map` and `f_flat_map` to achieve that.

Reply:
I have a question, can we upload as many files as we want with the same upload_request? If so, then I can change it to instead of upload_one_file, but to upload_files.

Reply:
Based on the assumptions(one upload request could upload as many files as wanted and upload the same file could be uploaded by chunks in threads), I implemented it as:
request upload id, blocking => upload files in threads, every file upload is wrapped in a future, doesn't block each other, also upload the same file in threads as well, each part of the uploads doesn't block each other => waiting for upload to be done and then import, blocking => delete upload request, non-blocking, return a future. 
Is this Ok?

Reply:
One upload request can only upload one file.

Reply:
Hm, then further change's needed.